### PR TITLE
Create a build job and add check for TODO and STOPSHIP comments

### PR DESCRIPTION
**Pull request by matir91**
### Description:
Right now if you have a lint issue, for example an image is missing, the deploy will not be done. However, it would be nice review it without the image.

I've also added a check for todo fixme, and stopship comments

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -9,7 +9,28 @@ on:
...
+        key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-gems-
+    - name: Install Ruby dependencies
+      run: |
+        bundle config path vendor/bundle
+        bundle install --jobs 4 --retry 3
+    - name: Build Jekyll
+      run: bundle exec jekyll build --trace --strict_front_matter
   build_and_check:
```
Comment (bryant1410) [inline .github/workflows/main.yml:32]:
```suggestion
  lint:
```

?

Comment (matir91) [reply .github/workflows/main.yml:32]: I didn't change it in order to keep the required github job setup. Also it compiles, so for me it makes sense. What do you think? should I change it?

Comment (bryant1410) [reply .github/workflows/main.yml:32]: required github job setup?

It's a minor thing, so as you want

Comment (matir91) [reply .github/workflows/main.yml:32]: yes, it's required and I have not permission to change it. That's way I didn't change it.
 
image

**Pull request merged**